prog/test/postgres_resource.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::Test::PostgresResource < Prog::Test::Base
+  semaphore :destroy
+
+  def self.assemble(test_case, vm_host_id, postgres_test_project_id)
+    Strand.create_with_id(
+      prog: "Test::PostgresResource",
+      label: "start",
+      stack: [{
+        "postgres_test_project_id" => postgres_test_project_id,
+        "vm_host_id" => vm_host_id,
+        "test_case" => test_case
+      }]
+    )
+  end
+
+  label def start
+    hop_download_boot_image
+  end
+
+  label def download_boot_image
+    flavor = "-#{tests[test_case]["postgres_flavor"]}"
+    flavor = "" if flavor == "-standard"
+    image_name = "postgres#{tests[test_case]["postgres_version"]}#{flavor}-ubuntu-2204"
+    vm_host = VmHost[vm_host_id]
```

Comment:
I don't think we should pass around `vm_host_id`, just to use it to download boot image. There is already only one host in the E2E tests, we can directly download the boot image there. And even if we add more hosts, it is OK to download boot image to all of them.

I check the GithubRunner tests. It seems we also pass `vm_host_id` there. @enescakir, @velioglu any reason we decided to do it like that?

Reply:
Actually, why don't we download all necessary images while setting up the host?

Reply:
@byucesoy do you mean download it in `bin/ci`after the `hetzner_server_st` is ready or within `prog/test/hetzner_server.rb` itself, since it is only used for E2E tests?

Reply:
I can't recall the exact reason right now. @velioglu chose to download the image during the test, rather than when setting up the host. He might know more about this.

Reply:
I mean in `prog/test/hetzner_server.rb`

Reply:
Since downloading runner images takes few minutes, I downloaded it via test code to not download them if runner tests are not running. Do the same for specific host in case there will be more than 1 host in the future. If we will download them conditionally while setting up the host it is ok.

Reply:
I have added a new config called `e2e_boot_images.yml` which lists the boot images for different tests cases. In `./bin/ci`, this file is read and we pass whichever boot images are required for the list of test cases that are being run to `default_boot_images` arg in `Prog::Vm::HostNexus.assemble`.

Reply:
99% of the time, we run all the tests, right? I think it would be better to download all the necessary images all the time for the sake of simplicity (no new configs, no extra logic to fetch additional boot images).

We don't need to refactor runner tests I think (I'd prefer to, but that would require buy-in from runner team and also I don't want to increase the scope of this PR), but for Postgres, I think we can download the image all the time.

Reply:
We had a sync on this with @enescakir. Let's leave this as is. He will open a follow up PR that combines logic for all E2E tests.

Reply:
I created a draft PR for it.
https://github.com/ubicloud/ubicloud/pull/2642